// 举例UserSerive如何发布、调用
syntax = "proto3";

package kit_rpc;

option cc_generic_services = true;

message ResultCode
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

message LoginRequest
{
    bytes name = 1;
    bytes pwd = 2;
}

message LoginResponse
{
    ResultCode result = 1;
    bool success = 2;
}

message LogOutRequest
{
    bytes name = 1;
}

message LogOutResponse
{
    ResultCode result = 1;
    bool success = 2;
}

service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc LogOut(LogOutRequest) returns(LogOutResponse);
}